WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -2655,6 +2690,160 @@ namespace Plugin {
+                return false;
+            }
+            _displayConnectionProps->Register(this);
+            SYSLOG(Logging::Notification, (_T("Subscribed to HDR capabilities updates from DisplayInfo plugin.")));
+            return true;
+        }
+
+        void UnsubscribeHDRCapabilities()
+        {
+            if (_displayConnectionProps) {
+                _displayConnectionProps->Unregister(this);
+                _displayConnectionProps->Release();
+                _displayConnectionProps = nullptr;
+            }
+            _adminLock.Lock();
+            if (_hdrRefreshJob.IsValid()) {
+                Core::WorkerPool::Instance().Revoke(_hdrRefreshJob);
+                _hdrRefreshJob.Release();
+            }
+            _adminLock.Unlock();
+            if (_displayInfoPlugin) {
+                _displayInfoPlugin->Release();
+                _displayInfoPlugin = nullptr;
+            }
+            SYSLOG(Logging::Notification, (_T("Unsubscribed from HDR capabilities updates from DisplayInfo plugin.")));
+        }
+
+        void RefreshHDRSupport()
+        {
+            auto displayInfoPlugin = QueryDisplayInfoPlugin();
+            if (!displayInfoPlugin) {
+                return;
+            }
+            Exchange::IHDRProperties* hdrPropIface = displayInfoPlugin->QueryInterface<Exchange::IHDRProperties>();
+            if (!hdrPropIface) {
+                return;
+            }
+            Exchange::IHDRProperties::IHDRIterator* iter = nullptr;
+            hdrPropIface->STBCapabilities(iter);
+            if (!iter) {
+                hdrPropIface->Release();
+                return;
+            }
+
+            std::vector<Exchange::IHDRProperties::HDRType> stbCaps;
+            Exchange::IHDRProperties::HDRType value;
+            while (iter->Next(value)) {
+                stbCaps.push_back(value);
+            }
+            iter->Release();
+
+            hdrPropIface->TVCapabilities(iter);
+            if (!iter) {
+                hdrPropIface->Release();
+                return;
+            }
+
+            std::vector<Exchange::IHDRProperties::HDRType> commonCaps;
+            while (iter->Next(value)) {
+                if (std::find(stbCaps.begin(), stbCaps.end(), value) != stbCaps.end()) {
+                    commonCaps.push_back(value);
+                }
+            }
+            iter->Release();
+            hdrPropIface->Release();
+
+            bool isHDRSupported = false;
+            for (const auto& cap : commonCaps) {
+                if (cap > Exchange::IHDRProperties::HDRType::HDR_OFF) {
+                    isHDRSupported = true;
+                    break;
+                }
+            }
+
+            _adminLock.Lock();
+            _hdrSupported = isHDRSupported;
+            _adminLock.Unlock();
+        }
+
+        void UpdateHDRSettings(WebKitSettings* settings)
+        {
+            ASSERT(settings != nullptr);
+            _adminLock.Lock();
```

Comment:
Let's release the lock before calling webkit_settings_set_screen_supports_hdr, i.e.:

```
_adminLock.Lock();
bool hdrSupported = _hdrSupported;
_adminLock. Unlock();


```